File: src/types/resource.ts
// src/types/resource.ts

export type ResourceStatus = 'Available' | 'Busy' | 'On Leave';

export type Weekday = 'Mon' | 'Tue' | 'Wed' | 'Thu' | 'Fri' | 'Sat' | 'Sun';

export interface Therapist {
  id: string | number;
  name: string;
  status: ResourceStatus;
  specialty: string;
  role: 'Senior Therapist' | 'Therapist' | 'Junior Therapist';
  skills: string[];
  shifts: Weekday[];
}

export interface Room {
  id: string;
  name: string;
  capacity: number;
  facilities: string[];
  status: ResourceStatus;
}

export interface Material {
  id: string;
  name: string;
  quantity: number;
  unit: string;
  lowStockThreshold: number;
}

export interface ResourceState {
  therapists: Therapist[];
  rooms: Room[];
  materials: Material[];
  isLoading: boolean;
  error: string | null;
}

File: backend/app/models/resource.py
from typing import List, Union, Literal
from beanie import Document


Weekday = Literal["Mon", "Tue", "Wed", "Thu", "Fri", "Sat", "Sun"]
ResourceStatus = Literal["Available", "Busy", "On Leave"]


class Therapist(Document):
    thid: str
    name: str
    status: ResourceStatus
    specialty: str
    role: Literal["Senior Therapist", "Therapist", "Junior Therapist"]
    skills: List[str] = []
    shifts: List[Weekday] = []

    class Settings:
        name = "therapists"


class Room(Document):
    rid: str
    name: str
    capacity: int
    facilities: List[str] = []
    status: ResourceStatus

    class Settings:
        name = "rooms"


class Material(Document):
    mid: str
    name: str
    quantity: int
    unit: str
    lowStockThreshold: int

    class Settings:
        name = "materials"
